### PR TITLE
more `LocalDefId`s

### PR DESCRIPTION
**Pull request by lcnr**

### CODE REVIEW

**Comment by lcnr** [comment]
Rebased and ready for review

File: src/librustc_middle/query/mod.rs [hunk 1]
```diff
@@ -386,8 +386,15 @@ rustc_queries! {
             storage(ArenaCacheSelector<'tcx>)
         }
 
-        /// HACK: when evaluated, this reports a "unsafe derive on repr(packed)" error
-        query unsafe_derive_on_repr_packed(_: DefId) -> () {}
+        /// HACK: when evaluated, this reports a "unsafe derive on repr(packed)" error.
+        ///
+        /// Unsafety checking is executed for each method seperately, but we only want
```
Comment (petrochenkov) [inline src/librustc_middle/query/mod.rs:391]:
```suggestion
        /// Unsafety checking is executed for each method separately, but we only want
```

File: src/librustc_middle/query/mod.rs [hunk 2]
```diff
@@ -386,8 +386,15 @@ rustc_queries! {
...
+        /// to emit this error once per derive. As there are some impls with multiple
+        /// methods, we use a query for deduplication.
+        query unsafe_derive_on_repr_packed(key: LocalDefId) -> () {
+            desc { |tcx| "processing `{}`", tcx.def_path_str(key.to_def_id()) }
+            storage(ArenaCacheSelector<'tcx>)
```
Comment (petrochenkov) [inline src/librustc_middle/query/mod.rs:396]: This change seems unrelated to this PR?

Comment (lcnr) [reply src/librustc_middle/query/mod.rs:396]: Afaik we have to explicitly mention the storage and desc in case the key is not a `DefId`.

Comment (petrochenkov) [reply src/librustc_middle/query/mod.rs:396]: From what I see, most of queries taking `LocalDefId` don't specify storage.

**Comment by lcnr** [comment]
RFM I guess

**Comment by petrochenkov** [comment]
r=me after squashing the commits

**Comment by lcnr** [comment]
ok, done
